joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/astcreation/AstCreator.scala

```diff
@@ -94,14 +96,71 @@ class AstCreator(
         scope.pushNewScope(moduleScope)
         val block = blockNode(rootNode)
         scope.pushNewScope(BlockScope(block))
-        val statementAsts = rootNode.statements.flatMap(astsForStatement)
+        val statementAsts         = rootNode.statements.flatMap(astsForStatement)
+        val internalMethodRefAsts = methodRefNodesForInternalMethods()
+        val internalTypeRefAsts   = typeRefNodesForInternalDecls()
         scope.popScope()
-        val bodyAst = blockAst(block, statementAsts)
+        val bodyAst = blockAst(block, internalTypeRefAsts ++ internalMethodRefAsts ++ statementAsts)
         scope.popScope()
         methodAst(methodNode_, Seq.empty, bodyAst, methodReturn, newModifierNode(ModifierTypes.MODULE) :: Nil)
       }
       .getOrElse(Ast())
   }
+
+  private def methodRefNodesForInternalMethods(): List[Ast] = {
+    val typeNameForMethods = scope.surroundingTypeFullName
+      .map { x =>
+        x.split("[:]{2}").dropRight(1).mkString("")
+      }
+      .getOrElse(Defines.Undefined)
+
+    programSummary.namespaceToType
+      .filter(_._1 == typeNameForMethods)
+      .flatMap(_._2)
+      .filter(x => x.name.contains(":program"))
+      .flatMap(_.methods)
+      .filterNot(x => x.name.contains("<init>") || x.name.contains("<clinit>"))
```

Comment:
Use the `x2cpg.Defines` contains here

Reply:
Removed this `filterNot`, these methods are already being filtered out during the above `filter(_._1. ==)`